pinot-spi/src/main/java/org/apache/pinot/spi/stream/RowMetadata.java

```diff
@@ -39,5 +44,27 @@ public interface RowMetadata {
    * @return timestamp (epoch in milliseconds) when the row was ingested upstream
    *         Long.MIN_VALUE if not available
    */
-  long getIngestionTimeMs();
+  long getRecordTimestampMs();
```

Comment:
I disagree here. I don't think the semantics are clear. 

1. Ingestion time is vague. It can be interpreted as the time at which the record was ingested into Pinot table or the time at which the record was ingested into the upstream source. Looking at the implementation today, it seems to refer to the latter case. So , recordtimestamp is more accurate as it is the timestamp which comes with the incoming stream record.  
2. Oddly enough, this interface is named `RowMetadata`. However, a row is usually an entity in a table. It is not typically referred to the data in a streaming source. We need to start making the distinction more clear in Pinot.

Reply:
What is the semantic that you want here? Time at which the record was ingested into the stream? Time at which the record was generated upstream? Or some other?
In other words, in Pinot, if I write an expression `(System.currentTimeMillis() - getRecordTimeStampMs())` what do you expect this value to signify (independent of any underlying stream, of course)?

As a side note, please modify the javadoc with the answer to the q above.

Reply:
> What is the semantic that you want here?

Time at which the record was ingested into the stream. 

> Time at which the record was generated upstream

This would be event time and it is possible that the event time is used as the record timestamps (this is possible in Kafka by allowing record to use producer's timestamp). However, from Pinot's point of view, we only care about how soon a record became available for consumption. 

> In other words, in Pinot, if I write an expression (System.currentTimeMillis() - getRecordTimeStampMs()) what do you expect this value to signify (independent of any underlying stream, of course)?

I want this expression `System.currentTimeMillis() - getRecordTimestampMs()` to indicate the message availability for consumption since the time it became available in source. It should _tend towards 0_ when the consumer is fully caught up the upstream data

Reply:
I have already updated the javadoc for `RowMetadata#getRecordTimestampMs` . Let me know if you need more clarity or are referring to some other javadoc you want me to update.

Reply:
> > What is the semantic that you want here?
> 
> Time at which the record was ingested into the stream.

And hence the API getRecordIngestionTime()

Reply:
The API will be used by folks writing plugins for a new stream.  Imagine if you are writing a new stream, and need to implement getRecordTimeStamp() but your stream does not provide any field called RecordTimeStamp (but provides other ways to get the ingestion time).

And hence my suggestion.

I will let you decide.

Reply:
I was one of the folks writing plugin for a new stream and it wasn't clear to me :) 

I am fine changing it to `recordIngestionTime`. Previously, it was just "ingestionTime" and it was within "RowMetadata". So, altogether, it didn't make sense to me.

Reply:
renamed!